### PR TITLE
Change account color picker

### PR DESCRIPTION
**Pull request by cketti**
- Switch from HoloColorPicker to the one provided by the preferences library
- Use colors from `MaterialColors` for account colors

k9mail_account_color_picker

### CODE REVIEW

**Comment by ByteHamster** [comment]
I vote for sorting the colors, so they form kind of a rainbow. Currently, the order seems to be quite random.

**Comment by cketti** [comment]
The sorting feature of the color picker component leads to this:

k9mail_account_color_picker_sorted

@ByteHamster Do you like it better or do you want to give it a shot sorting the colors yourself?

File: app/core/src/main/res/values/arrays_account_settings_values.xml [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+        <item>0xFFFB8C00</item>
+        <item>0xFFE53935</item>
+        <item>0xFF8E24AA</item>
+        <item>0xFFD81B60</item>
+        <item>0xFF5E35B1</item>
+        <item>0xFF3949AB</item>
+        <item>0xFF00ACC1</item>
+        <item>0xFF00897B</item>
+        <item>0xFF7CB342</item>
+        <item>0xFFF4511E</item>
```
Comment (ByteHamster) [inline app/core/src/main/res/values/arrays_account_settings_values.xml]: Suggestion:


Not totally happy with it but I think it looks a bit more "ordered"

```
        <item>0xFFFB8C00</item>
        <item>0xFFF4511E</item>
        <item>0xFFE53935</item>
        <item>0xFFD81B60</item>
        <item>0xFF8E24AA</item>
        <item>0xFF5E35B1</item>
        <item>0xFF3949AB</item>
        <item>0xFF039BE5</item>
        <item>0xFF00ACC1</item>
        <item>0xFF7CB342</item>
        <item>0xFF388E3C</item>
        <item>0xFF00897B</item>
```

Comment (cketti) [reply app/core/src/main/res/values/arrays_account_settings_values.xml]: > Another possibility would be to add some more blue colors and add a fourth line of blue shades.

Sounds good.

**Comment by cketti** [comment]
The original idea was to replace the five existing Holo account colors with material variants. Then I added a couple of other colors to provide a greater selection.

What we now have to figure out is how we want to select the color for newly created accounts. 
Use a certain order, i.e. always make the first account color X? Should this be the same order in which the colors are displayed in the color picker? Should we aim for default account colors to be as distinct as possible and only when users have lots of accounts use a color that is similar to an account color already in use?
Should we just select a random unused color?

File: app/core/src/main/res/values/arrays_account_settings_values.xml (hunk 1 above)
Comment (ByteHamster) [reply app/core/src/main/res/values/arrays_account_settings_values.xml]: How about those colors? The last one does not fit 100% but I think it is good to have at least one color that is really neutral.





```
        <item>0xFFFFB300</item> <!-- Amber 600 -->
        <item>0xFFFB8C00</item> <!-- Orange 600 -->
        <item>0xFFF4511E</item> <!-- Deep orange 600-->
        <item>0xFFE53935</item> <!-- Red 600 -->

        <item>0xFFC0CA33</item> <!-- Lime 600 -->
        <item>0xFF7CB342</item> <!-- Light green 600 -->
        <item>0xFF388E3C</item> <!-- Green 700 -->
        <item>0xFF00897B</item> <!-- Teal 600 -->

        <item>0xFF00ACC1</item> <!-- Cyan 600 -->
        <item>0xFF039BE5</item> <!-- Light blue 600 -->
        <item>0xFF1976D2</item> <!-- Blue 700 -->
        <item>0xFF3949AB</item> <!-- Indigo 600 -->

        <item>0xFFE91E63</item> <!-- Pink 500 -->
        <item>0xFF8E24AA</item> <!-- Purple 600-->
        <item>0xFF5E35B1</item> <!-- Deep purple 600-->
        <item>0xFF455A64</item> <!-- Blue gray 700 -->
```

**Comment by ByteHamster** [comment]
> What we now have to figure out is how we want to select the color for newly created accounts.
Use a certain order, i.e. always make the first account color X?

I think choosing a first color is a good idea. Most users will probably never change the color, so it should be a nice one ;) Not sure which one though. Maybe the first color could be red600 because the icon uses a red-ish color? With that, I see the problem that gmail also uses red as its main color. My personal favorite is blue700 but I can't give any reason why I like the color so much.

> Should this be the same order in which the colors are displayed in the color picker?

No, I don't think so. If the colors are more or less sorted in the picker, one would get many shades of one color. Maybe having the first one fixed and selecting the others at random could work. I assume that users with multiple email addresses are already tech-savvy enough that they go and look for a color setting if they don't like the combination.

**Comment by cketti** [comment]
I picked three colors (blue 700, pink 500, amber 600) I think work reasonably well together as "default colors". After that colors are randomly picked.

k9mail_default_account_colors

**Pull request merged**